src/components/DateTimePicker/index.js

```diff
@@ -38,6 +38,7 @@ function PickerComponent(props, ref) {
     } = props;
 
     const inputRef = useRef();
+    const modalId = (id && `${id}_modal`) || undefined;
```

Comment:
here you only need:
```const modalId = id && `${id}_modal`;```
when id is undefined it will return the id value (undefined)
otherwise will return  `${id}_modal`